### PR TITLE
CRYP-98: implement frontend navbar

### PR DESCRIPTION
**Pull request by narroarrow**
- Added styling to navbar 
- Styling based on: https://www.figma.com/file/Igge56wzf6UAWZwFcCzglv/Handoff?node-id=5%3A5809
- Note: Added custom house icon from fontawesome rather than using TabBarIcon function. Can easily be changed to just use TabBarIcon if that is preferred.

 image

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,8 +1,10 @@
 {
```
Comment (jason-gerard) [inline tsconfig.json:1]: This file shouldn't be changed

**Comment by Andre-Ibrahim** [comment]
Can you please put a screenshot of the mobile screen in the description

**Review by domsec — COMMENTED** [review]
@jason-gerard we should probably take some time once all features are implemented and look into refactoring the way we import the icons. There must be a way we can import all related icon files and avoid having to import each one as needed like we currently are. What do you think?

File: packages/client/navigation/index.tsx [hunk 2]
```diff
@@ -56,6 +58,23 @@ function BottomTabNavigator() {
             initialRouteName="HomeScreen"
             screenOptions={{
                 tabBarActiveTintColor: Colors[colorScheme].tint,
+                tabBarLabelPosition: "below-icon",
+                tabBarInactiveTintColor: "#404040",
```
Comment (domsec) [inline packages/client/navigation/index.tsx:62]: Replace with `Colors[colorScheme].tabIconDefault` and in `packages\client\constants\Colors.ts` file replace `tabIconDefault: "#ccc",` with `tabIconDefault: "#404040",

File: tsconfig.json (hunk 1 above)
Comment (narroarrow) [reply tsconfig.json:1]: Accidently ran expo on root folder, removed any changes

File: packages/client/components/icons/houseNav.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { IconDefinition, IconName, IconPrefix } from "@fortawesome/fontawesome-svg-core";
+
+export const houseNav: IconDefinition = {
```
[the remark this answers is not in the document]
Comment (narroarrow) [reply packages/client/components/icons/houseNav.tsx]: Done

File: packages/client/components/icons/houseNav.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        512,
+        // Aliases (not needed)
+        [],
+        // Unicode as hex value (not needed)
+        "",
+        // SVG path data
+        "M575.8 255.5c0 18-15 32.1-32 32.1h-32l.7 160.2c0 2.7-.2 5.4-.5 8.1V472c0 22.1-17.9 40-40 40H456c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1H416 392c-22.1 0-40-17.9-40-40V448 384c0-17.7-14.3-32-32-32H256c-17.7 0-32 14.3-32 32v64 24c0 22.1-17.9 40-40 40H160 128.1c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2H104c-22.1 0-40-17.9-40-40V360c0-.9 0-1.9 .1-2.8V287.6H32c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z",
+    ],
+    iconName: "fa-house" as IconName,
+    prefix: "house" as IconPrefix,
```
[the remark this answers is not in the document]
Comment (narroarrow) [reply packages/client/components/icons/houseNav.tsx]: Done

File: packages/client/navigation/index.tsx [hunk 5]
```diff
@@ -70,15 +89,16 @@ function BottomTabNavigator() {
...
+                    tabBarIcon: ({ color }) => <FontAwesomeIcon icon={houseNav} color={color} size={30} />,
                 }}
             />
             <BottomTab.Screen
                 name="SettingsScreen"
                 component={SettingsScreen}
                 options={{
                     title: "Settings",
-                    tabBarIcon: ({ color }) => <TabBarIcon name="code" color={color} />,
+                    tabBarIcon: ({ color }) => <TabBarIcon name="bars" color={color} />,
```
[the remark this answers is not in the document]
Comment (narroarrow) [reply packages/client/navigation/index.tsx]: Done

File: packages/client/navigation/index.tsx [hunk 6]
```diff
@@ -56,6 +58,23 @@ function BottomTabNavigator() {
             initialRouteName="HomeScreen"
             screenOptions={{
                 tabBarActiveTintColor: Colors[colorScheme].tint,
```
[the remark this answers is not in the document]
Comment (narroarrow) [reply packages/client/navigation/index.tsx]: Done

File: packages/client/navigation/index.tsx [hunk 7]
```diff
@@ -56,6 +58,23 @@ function BottomTabNavigator() {
...
+                    paddingHorizontal: 8,
+                },
+
+                tabBarLabelStyle: {
+                    height: 34,
+                    lineHeight: 16,
+                    fontSize: 12,
+                    fontWeight: "400",
+                    letterSpacing: 0.5,
+                },
```
[the remark this answers is not in the document]
Comment (narroarrow) [reply packages/client/navigation/index.tsx]: Done

File: packages/client/navigation/index.tsx [hunk 8]
```diff
@@ -56,6 +58,23 @@ function BottomTabNavigator() {
             initialRouteName="HomeScreen"
             screenOptions={{
                 tabBarActiveTintColor: Colors[colorScheme].tint,
+                tabBarLabelPosition: "below-icon",
+                tabBarInactiveTintColor: "#404040",
+
+                tabBarStyle: {
+                    height: 78,
+                    paddingTop: 8,
+                    paddingHorizontal: 8,
```
[the remark this answers is not in the document]
Comment (narroarrow) [reply packages/client/navigation/index.tsx]: Done

File: packages/client/navigation/index.tsx [hunk 9]
```diff
@@ -70,15 +89,16 @@ function BottomTabNavigator() {
                 component={HomeScreen}
                 options={{
                     title: "Home",
-                    tabBarIcon: ({ color }) => <TabBarIcon name="code" color={color} />,
+                    // Note: Trying out the home button icon imported from FontAwesome
+                    tabBarIcon: ({ color }) => <FontAwesomeIcon icon={houseNav} color={color} size={30} />,
```
[the remark this answers is not in the document]
Comment (narroarrow) [reply packages/client/navigation/index.tsx]: Done

File: packages/client/navigation/index.tsx (hunk 2 above)
Comment (narroarrow) [reply packages/client/navigation/index.tsx]: Done

**Comment by narroarrow** [comment]
image

**Review by jason-gerard — APPROVED** [review]
Make sure to revert the package.json, yarn.lock, and tsconfig.json before merging.

**Pull request merged**